nova/compute/manager.py

```diff
@@ -8867,3 +8867,29 @@ def _cleanup_expired_console_auth_tokens(self, context):
     def sync_server_group(self, context, sg_uuid):
         """Calls the driver to update the server-group in the backend"""
         self.driver.sync_server_group(context, sg_uuid)
+
+    @periodic_task.periodic_task(spacing=CONF.instance_flavor_update_interval)
+    def _update_instances_flavor_info(self, context):
+        instances = objects.InstanceList.get_by_host(context, self.host,
+                                                     expected_attrs=['flavor'],
+                                                     use_slave=True)
+        if not instances:
+            return
+        flavors = objects.FlavorList.get_all(context)
+
+        def _find_flavor(instance):
+            for flavor in flavors:
+                if instance.flavor and instance.instance_type_id == flavor.id:
```

Comment:
I don't think we can rely on the `instance_type_id`, because for some operations we had to delete the flavor and create a new one - which makes the `flavor.id` change. Instead, we have to rely on the name of the flavor being the same - or being an alias.